### DISCUSSION TITLE
Single connection multiple rooms is possible ?

### DISCUSSION THREAD

**Discussion by arun6894** [miguelgrinberg/python-socketio#1133] [Q&A] [answered]
@miguelgrinberg  Can we use a single sid to enter into two different rooms?

For eg.,   sid = 'sfsfs32dssfsd3243sf'  and a client entered room A with this.

Later with the help of client if we are receiving the same sid and we will enter the sid into another room B.

Now we entered the same sid in two different rooms Room A and Room B.

when we tried to emit to Room A does the message will send to Room B as well as we have the same sid?

-------------------------------------------------------------------------------------------------------------

I have one more question. Is it a good practice to add Kafka module initialization in __init__ to the socket Namespace class like below or create an object every time when we need to produce ? 

```
class MycustomNamespace(socketio.Namespace):

    def __init__(self, namespace=None):

        super(MycustomNamespace, self).__init__(namespace=namespace)
        self.kafka = service.kafka_manager.SocketService()
```

**Comment by miguelgrinberg** [comment, marked as answer]
Yes, a client can enter any number of rooms, there is no limit. Regarding the second question, It is fine to initialize your kafka service in the namespace constructor.
